spring-pulsar/src/main/java/org/springframework/pulsar/core/CachingPulsarProducerFactory.java

```diff
@@ -193,7 +197,8 @@ static class ProducerCacheKey<T> {
 			Assert.notNull(schema, () -> "'schema' must be non-null");
 			Assert.notNull(topic, () -> "'topic' must be non-null");
 			this.schema = schema;
-			this.schemaHash = SchemaHash.of(this.schema);
+			this.schemaHash = (schema instanceof AutoProduceBytesSchema) ? AUTO_PRODUCE_SCHEMA_HASH
```

Comment:
This is necessary as any call to `AutoProduceSchema.getSchemaInfo()` will throw RuntimeException on the client until after something has been sent and validated w/ the target topic (i.e a schema is initialized on it). In our case, we don't care about schema info, the producer can be used for any `byte[]` messages that the user wants to be validated against the target topic schema and this cache key will do that.